platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -160,18 +169,23 @@ public void run() {
          * @param location Location of the DER-encoded CRL file that should be used to
          *                 check certificate revocation.
          */
-        public void setCrlLocation(String location) {
-            CRL current = crlCache.get();
-            if (location == null) {
-                LOGGER.info("CRL property in {} is not set. Certs will not be checked against a CRL",
-                        encryptionPropertiesLocation);
-                crlCache.set(null);
-            } else {
-                CRL crl = createCrl(location);
-                if (crl != null) {
-                    crlCache.set(crl);
-                    LOGGER.info("CRL has been updated from {}.", location);
+        void setCrlLocation(String location) {
+            lock.lock();
```

Comment:
@stustison This reentrant lock will prevent concurrent access from the scheduled updater and the explicit call to set the location; however...never mind. I was going to say that anyone explicitly setting the `crlLocation` would have their value replaced on the next scheduled check/next day depending on the `nextUpdate` value of the crl, but I see that the wrapping class's `setCrlLocation` method - is only ever invoked by unit test.